aidial_adapter_openai/gpt4_multi_modal/truncate.py

```diff
@@ -0,0 +1,63 @@
+from typing import List, Set
+
+from aidial_sdk.exceptions import (
+    TruncatePromptSystemAndLastUserError,
+    TruncatePromptSystemError,
+)
+
+from aidial_adapter_openai.gpt4_multi_modal.transformation import (
+    MessageTransformResult,
+    TruncateTransformedMessagesResult,
+)
+
+
+def truncate_transformed_messages(
+    transformations: List[MessageTransformResult], max_prompt_tokens: int
```

Comment:
It's basically a copy-paste of `aidial_adapter_openai.utils.tokens:truncate_prompt`.
Why not reuse with a sufficient generalization?

Reply:
Good catch
Generalized this in one function